traits/traits_listener.py

```diff
@@ -341,7 +341,7 @@ def register ( self, new ):
         """
         # Make sure we actually have an object to set listeners on and that it
         # has not already been registered (cycle breaking):
-        if (new is None) or (new is Undefined) or (new in self.active):
+        if not isinstance(new, HasTraits) or new in self.active:
```

Comment:
why do you remove two conditions from the test? The code does not support `None` anymore with this change. It will also cause issues if new is `Undefined`